src/Worker/Grpc/Logs.cs

```diff
@@ -36,8 +35,8 @@ static partial class Logs
         [LoggerMessage(EventId = 14, Level = LogLevel.Debug, Message = "{instanceId}: Sending {successOrFailure} response for '{name}#{taskId}' activity with {sizeInBytes} bytes of output data.")]
         public static partial void SendingActivityResponse(this ILogger logger, string successOrFailure, string name, int taskId, string instanceId, int sizeInBytes);
 
-        [LoggerMessage(EventId = 20, Level = LogLevel.Error, Message = "Unexpected error in handling of instance ID '{instanceId}'. Details: {details}")]
-        public static partial void UnexpectedError(this ILogger logger, string instanceId, string details);
+        [LoggerMessage(EventId = 20, Level = LogLevel.Error, Message = "Unexpected error in handling of instance ID '{instanceId}'.")]
+        public static partial void UnexpectedError(this ILogger logger, Exception ex, string instanceId);
```

Comment:
I'm curious to understand this change better: is ther especial handling for `Exception` objects in the log generator?

Reply:
Yup! https://learn.microsoft.com/en-us/dotnet/core/extensions/logger-message-generator#log-method-anatomy

> As a general rule, the first instance of `ILogger`, `LogLevel`, and `Exception` are treated specially in the log method signature of the source generator. Subsequent instances are treated like normal parameters to the message template

This will now pass to the explicit `Exception` parameter on `ILogger.Log`